src/tarski/reachability/clingo_wrapper.py

```diff
@@ -40,22 +40,29 @@ def run_clingo(lp):
 
     raise ExternalCommandError(f"Unknown Gringo error. Gringo exited with code {retcode}. Full error log: {errlog}")
 
+def parse_model(*, filename=None, content=None, symbol_mapping):
+    if filename and not content:
+        with open(filename, "r") as f:
+            return _parse_model((l for l in f), symbol_mapping)
```

Comment:
Would this make it an iterator, or build the full list before calling the method? Might seem weird, but would just passing in `f` work? It's treated as `lines` in `_parse_model`, and may well be just the iterator you're looking for...

Reply:
The `(...)` syntax creates a generator, but you're totally right that passing in `f` should work as well;
likewise, passing in `content.splitlines()` below should work too.